src/dbnode/encoding/proto/common.go

```diff
@@ -165,18 +171,27 @@ func newCustomFieldState(fieldNum int, fieldType customFieldType) customFieldSta
 
 // TODO(rartoul): Improve this function to be less naive and actually explore nested messages
 // for fields that we can use our custom compression on: https://github.com/m3db/m3/issues/1471
-func customFields(s []customFieldState, schema *desc.MessageDescriptor) []customFieldState {
+func customFields(s []customFieldState, protoFields []int32, schema *desc.MessageDescriptor) ([]customFieldState, []int32) {
+	fields := schema.GetFields()
 	numCustomFields := numCustomFields(schema)
+	numProtoFields := len(fields) - numCustomFields
+
 	if cap(s) >= numCustomFields {
 		s = s[:0]
```

Comment:
nit: Do you need to nil out state on elements inside `customFieldState`?  If there's any pointers there they will still point to things and held onto by the garbage collector.

i.e.
```
for i := range s {
  s[i] = customFieldState{}
}
```

Reply:
I always forget to do that, thanks